cpp/src/text/vocabulary_tokenize.cu

```diff
@@ -86,10 +86,17 @@ struct vocab_equal {
     return lhs == rhs;  // all rows are expected to be unique
   }
   // used by find
+#pragma nv_diagnostic push
+#pragma nv_diag_suppress 177
```

Comment:
Can you put a comment here on what this is suppressing?
Perhaps it can be removed with a future version of the compiler?

Reply:
Good point. Comments added as requested.

Once the cuco bump PR merged into `rapids-cmake`, I will open another PR removing all redundant overloads together with the pragmas added in this PR. I've also tested `[[maybe_unused]]` locally but it couldn't get rid of the warning.